subprojects/testing-base/src/testFixtures/groovy/org/gradle/testing/AbstractTestFrameworkIntegrationTest.groovy

```diff
@@ -150,11 +157,114 @@ abstract class AbstractTestFrameworkIntegrationTest extends AbstractIntegrationS
         testResult.assertTestClassesExecuted('SomeTest', 'NewTest')
     }
 
+    def "honors test case filter from --tests flag"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.$passingTestCaseName"
+
+        then:
+        testResult.assertTestClassesExecuted('SomeOtherTest')
+        testResult.testClass('SomeOtherTest').assertTestPassed(passingTestCaseName)
+    }
+
+    def "honors test suite filter from --tests flag"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.*"
+
+        then:
+        testResult.assertTestClassesExecuted('SomeOtherTest')
+        testResult.testClass('SomeOtherTest').assertTestPassed(passingTestCaseName)
+    }
+
+    def "reports when no matching methods found"() {
+        given:
+        createPassingFailingTest()
+
+        //by command line
+        when: fails(testTaskName, "--tests", "${testSuite('SomeTest')}.missingMethod")
+        then: failure.assertHasCause("No tests found for given includes: [${testSuite('SomeTest')}.missingMethod](--tests filter)")
+
+        //by build script
+        when:
+        buildFile << "${testTaskName}.filter.includeTestsMatching '${testSuite('SomeTest')}.missingMethod'"
+        fails(testTaskName)
+        then: failure.assertHasCause("No tests found for given includes: [${testSuite('SomeTest')}.missingMethod](filter.includeTestsMatching)")
+    }
+
+    def "task is out of date when --tests argument changes"() {
+        String testTaskPath = ":$testTaskName"
```

Comment:
I don't get why passing this string directly to the `contains` method below doesn't work. It clearly shows the `skippedTasks` list containing `:xcTest` and et are building the exact same string. Can the code reviewer double check this?

Reply:
This is probably a difference between `GString` and a `String`.

`result.skippedTasks` is a `Set<String>` so `result.skippedTasks.contains(":$testTaskName")` will be false because `"$testTaskName"` is a `GString`.  Declaring this as a String is coercing it into a String.

I think you can instead do:

`result.assertTaskSkipped(":$testTaskName")`

Reply:
Good point, I changed it.